src/pages/ExpiringResults/index.js

```diff
@@ -0,0 +1,204 @@
+              isInline
+              style={{ marginBottom: '8px' }}
+              onClick={() => this.navigateToRunResult(text[0])}
+            >
+              {text[0]}
+            </Button>
+            <br />
+            <Text component={TextVariants.p} className={styles.subText}>
+              <span className={styles.label}>{text[1]}</span>
+            </Text>
+          </div>
+        ),
+      },
+      {
+        title: 'End Time',
+        dataIndex: 'end',
+        key: 'end',
+      },
+      {
+        title: 'Scheduled for deletion on',
+        dataIndex: 'deletion',
+        key: 'deletion',
+        render: text => {
+          const deleteDate = moment(new Date(Date.parse(text)));
+          const currDate = moment(new Date());
+          const remainingDays = deleteDate.diff(currDate, 'days');
+          if (remainingDays > 45) {
+            return (
+              <div>
+                <Text>
+                  {moment(text)
+                    .add(7, 'days')
+                    .format('YYYY-MM-DDTHH:mm:ss:SSSSSS')}
+                </Text>
+                <Progress
+                  min={0}
+                  max={expirationLimit}
+                  value={expirationLimit - remainingDays}
+                  size={ProgressSize.sm}
+                  measureLocation={ProgressMeasureLocation.none}
+                />
+              </div>
+            );
+          }
+          return (
+            <div>
+              <span>
+                {moment(text)
+                  .add(7, 'days')
+                  .format('YYYY-MM-DDTHH:mm:ss:SSSSSS')}
+              </span>
+              <Progress
+                min={0}
+                max={expirationLimit}
+                value={expirationLimit - remainingDays}
+                size={ProgressSize.sm}
+                variant={ProgressVariant.danger}
+                measureLocation={ProgressMeasureLocation.none}
+              />
+            </div>
+          );
+        },
+      },
+      {
+        title: 'Status',
+        dataIndex: 'status',
+        key: 'status',
+      },
+      {
+        title: '',
+        dataIndex: 'fav',
+        key: 'fav',
+        render: (value, row) => {
+          let isFavorite = false;
+          favoriteControllers.forEach(item => {
+            if (item.key === row.key) {
+              isFavorite = true;
+            }
+          });
+          if (isFavorite) {
+            return (
+              <a onClick={e => this.removeControllerFromFavorites(e, null, row)}>
+                <Icon type="star" theme="filled" />
+              </a>
+            );
+          }
+          return (
+            <a onClick={e => this.favoriteRecord(e, null, row)}>
+              <Icon type="star" />
+            </a>
+          );
+        },
+      },
+      {
+        title: '',
+        dataIndex: 'action',
+        key: 'action',
+        render: (value, row) => {
+          return (
+            <div>
```

Comment:
Take a look at the `Menu` component from PatternFly for handling table actions.

Reply:
waiting on #124 for incorporation PR table actions